aws-android-sdk-core/src/main/java/com/amazonaws/util/json/JsonUtils.java

```diff
@@ -141,6 +141,87 @@ public static AwsJsonWriter getJsonWriter(Writer out) {
         return jsonToMap(new StringReader(json));
     }
 
+    /**
+     * Convenient method to convert a JSON string to a map. Arrays
+     * are not discarded, they are read into the value as a string.
```

Comment:
Huh. The behavior of `jsonToMap` is to discard arrays? That seems weird. I guess if we update that method, it'd break other things, is that right? 

Instead of squashing the list, and instead of using a `Map`, it feels like we should be recursing down the JSON, and calling the unmarshaller on everything we see, and returning a propertly modeled AWS SDK model class of some kind. Like, `FooServiceSpecificException`, that has a code-generated unmarshaller, that knows how to handle all of the fields in its model?

Reply:
@jamesonwilliams  It seems that a list property inside an exception is unprecedented so far, which suggests that it likely does not break anything. It appears that this code was written with an assumption that an error response will not contain any lists, which worked until now.